src/DoctrineMongoODMModule/Paginator/Adapter/DoctrinePaginator.php

```diff
@@ -19,7 +20,7 @@ class DoctrinePaginator implements AdapterInterface
     /**
      * Constructor
      */
-    public function __construct(Cursor $cursor)
+    public function __construct(Iterator $cursor)
```

Comment:
This makes no sense on its own, the class is still depending it'll get a `Cursor` instance

Reply:
The CommandCursor does not extend Cursor so it cannot be used in the DoctrinePaginator, which is the point of this change.

Reply:
Yeah but for one in `count()` method there's a call to `$this->cursor->getBaseCursor()` - from `\Iterator`'s perspective there's no such method

Reply:
Alright.  I'll close this PR and mark the issue as will not address.